### ISSUE TITLE
Both D400 and D415 error messages with `--select ALL`

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [astral-sh/ruff#8469]
When selecting ALL rules or just D400 and D415, D415 should take precedence over D400 (or the reverse?) so that ruff prints a single error message instead of two messages:
* ```python
  """Documentation"""
  ```
* ```
  $ ruff --isolated --select D400,D415 /path/to/file.py
  /path/to/file.py:1:1: D400 First line should end with a period
  /path/to/file.py:1:1: D415 First line should end with a period, question mark, or exclamation point
  Found 2 errors.
  No fixes available (2 hidden fixes can be enabled with the `--unsafe-fixes` option).
  $ 
  ```
* ```
  $ ruff --version
  ruff 0.1.3
  $ 
  ```

**Comment by charliermarsh** [comment]
I agree this is non-ideal, but making rules dependent on whether other rules are enabled causes problems and breaks some assumptions within Ruff. For example, if you have `# noqa: D415` on that line, but then run `ruff check --select D400`, `D400` would now be triggered instead of `D415` (since `D415` isn't active), and the violation would be raised. If you added a `# noqa: D400, D415`, then the `# noqa: D400` portion would appear unused and get flagged as an unused `# noqa`.

I think I'd prefer to solve this via rule recategorization. Like, this seems more appropriate as a setting on the rule than as two separate rules.

**Comment by zanieb** [comment]
Let's group this into https://github.com/astral-sh/ruff/issues/1774
